### ISSUE TITLE
UIKitView is not shadowed by Dialog

### ISSUE THREAD

**Issue by KevinnZou** [JetBrains/compose-multiplatform#3881]
**Describe the bug**
UIKitView is not shadowed by Dialog. It works well on Android and iOS.
https://github.com/KevinnZou/compose-webview-multiplatform/issues/34

**Affected platforms**
Select one of the platforms below:
- iOS

**Versions**
- Kotlin version*: 1.9.10
- Compose Multiplatform version*: 1.5.1

**To Reproduce**
commonMain:
```
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DialogSample() {
    var showDialog by remember { mutableStateOf(false) }
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(text = "WebView Sample") },
                actions = {
                    IconButton(onClick = { showDialog = true }) {
                        Icon(Icons.Default.Add,"Dialog")
                    }
                }
            )
        }
    ) {
        NativeView()
    }
    if (showDialog) {
        Dialog(onDismissRequest = { showDialog = false }) {
            Box(modifier = Modifier.background(Color.LightGray).size(200.dp)){
                Text(text = "Hello World!")
            }
        }
    }
}

@Composable
expect fun NativeView()
```

iOSMain:
```
@OptIn(ExperimentalForeignApi::class)
@Composable
actual fun NativeView() {
    UIKitView(
        factory = {
            UITextView().apply {
                this.text = Array(100) { "$it" }.joinToString("\n")
            }
        },
        modifier = Modifier.fillMaxSize()
    )
}
```
androidMain:
```
@Composable
actual fun NativeView() {
    AndroidView(
        factory = { context ->
            TextView(context).apply {
                this.text = Array(100) { "$it" }.joinToString("\n")
            }
        },
    )
}
```

**Expected behavior**
![截屏2023-10-30 10 05 09](https://github.com/JetBrains/compose-multiplatform/assets/43654611/b9d54b03-ecf1-45b0-b9b9-a4259389a6bf)

**Screenshots**
![截屏2023-10-30 09 51 40](https://github.com/JetBrains/compose-multiplatform/assets/43654611/dc867ed1-f721-4e4d-8347-a56bac1b9aa3)

**Additional context**

**Comment by dima-avdeev-jb** [comment]
Thanks, added reproducer with your code samples here https://github.com/dima-avdeev-jb/issue-3879-dialog-shadow

**Comment by dima-avdeev-jb** [comment]
Good news, it is already fixed in version Compose 1.5.10-rc02

<img width="489" alt="image" src="https://github.com/JetBrains/compose-multiplatform/assets/99798741/3faa5775-a5a0-4e0e-be23-010617b55b96">

**Comment by MatkovIvan** [comment]
That's probably about #812 (merged before 1.5.10-beta01)

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3881">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
